apps/challenges/views.py

```diff
@@ -340,3 +361,301 @@ def challenge_phase_split_list(request, challenge_pk):
+    except yaml.YAMLError as exc:
+        response_data = {
+            'error': exc
+            }
+        return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
+
+    # Check for evaluation script path in yaml file.
+    evaluation_script = yaml_file_data['evaluation_script']
+    if evaluation_script:
+        evaluation_script_path = join(BASE_LOCATION,
+                                      unique_folder_name,
+                                      extracted_folder_name,
+                                      evaluation_script)
+    else:
+        response_data = {
+            'error': 'There is no key for evaluation script in yaml file.'\
+                      'Please add a key and then try uploading it again!'
+            }
+        return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
+
+    # Check for evaluation script file in extracted zip folder.
+    if isfile(evaluation_script_path):
+        with open(evaluation_script_path, 'rb') as challenge_evaluation_script:
+            challenge_evaluation_script_file = ContentFile(challenge_evaluation_script.read(), evaluation_script_path)
+    else:
+        response_data = {
+            'error': 'No evaluation script is present in the zip file.'\
+                     'Please try uploading again the zip file after adding evaluation script!'
+            }
+        return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
+
+    # Check for test annotation file path in yaml file.
+    yaml_file_data_of_challenge_phases = yaml_file_data['challenge_phases']
+    for data in yaml_file_data_of_challenge_phases:
+        test_annotation_file = data['test_annotation_file']
+        if test_annotation_file:
+            test_annotation_file_path = join(BASE_LOCATION,
+                                             unique_folder_name,
+                                             extracted_folder_name,
+                                             test_annotation_file)
+        else:
+            response_data = {
+                'error': 'There is no key for test annotation file for'\
+                         'challenge phase {} in yaml file.'\
+                         'Please add a key and then try uploading it again!'.format(data['name'])
+                }
+            return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
+
+        if not isfile(test_annotation_file_path):
+            response_data = {
+                'error': 'No test annotation file is present in the zip file'\
+                         'for challenge phase {} Please try uploading'\
+                         'again the zip file after adding test annotation file!'.format(data['name'])
+            }
+            return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
+
+    # Check for challenge image in yaml file.
+    image = yaml_file_data['image']
+    if image.endswith('.jpg') or image.endswith('.jpeg') or image.endswith('.png'):
```

Comment:
Please make sure that we mention in our documentation that the logo can be of `jpg/jpeg/png` type only.